src/Util/UpdateChecker.php

```diff
@@ -0,0 +1,70 @@
+<?php
+namespace SilverStripe\Upgrader\Util;
+
+use Composer\Semver\Semver;
+use Humbug\SelfUpdate\Updater;
+use InvalidArgumentException;
+use SilverStripe\Upgrader\CodeCollection\CodeChangeSet;
+use SilverStripe\Upgrader\CodeCollection\DiskCollection;
+use SilverStripe\Upgrader\CodeCollection\ItemInterface;
+use SilverStripe\Upgrader\Composer\ComposerInterface;
+use SilverStripe\Upgrader\Composer\SilverstripePackageInfo;
+use Symfony\Component\Console\Exception\LogicException;
+
+/**
+ * Class to help check if the current PHAR executable is outdated.
+ */
+class UpdateChecker
+{
+
+    /**
+     * @var Updater $updater
+     */
+    private static $updater;
+
+    /**
+     * Build an updater object.
+     * @param string $version
+     * @return Updater
+     */
+    public static function getUpdater(string $version): Updater
+    {
+        if (self::$updater) {
+            return self::$updater;
+        }
+
+        $checkSignature = false;
+        self::$updater = new Updater(\Phar::running(false), $checkSignature, Updater::STRATEGY_GITHUB);
+        /** @var GithubStrategy $strategy */
+        $strategy = self::$updater->getStrategy();
+        $strategy->setPharName('upgrade-code.phar');
+        $strategy->setPackageName('silverstripe/upgrader');
+        $strategy->setCurrentLocalVersion($version);
+        return self::$updater;
+    }
+
+    /**
+     * Get the latest upgrader version available if the message has not been displayed or false otherwise.
+     *
+     * This save the value in memory. So you don't always get the message.
```

Comment:
`This saves the value in memory, so you don't always get the message.` instead of `This save the value in memory. So you don't always get the message.`

Reply:
Also, it is not stored in memory.

Reply:
`sys_get_temp_dir()` on *nix based system will return a location in `/tmp`, so it will be in memory.

On `Win` system, it returns a location in `C:\Windows\temp`. I've added a timestamp check as well because the file won't be deleted on reboot.